build.wake

```diff
@@ -0,0 +1,416 @@
+  | runJob
+  | getJobOutputs
+
+def filterWakeAndDotFiles =
+  _
+  | filter (!matches `(.*/)*\..*` _.getPathName)
+  | filter (!matches `.*\.wake` _.getPathName)
+
+#
+# Build Environment
+#
+
+export tuple FreedomESDKBuildEnv =
+  export Resources: List String
+  export BuildDir: String
+
+export def defaultFreedomESDKResources =
+  "riscv-tools/2020.08.2",
+  "python/python/3.7.1",
+  Nil
+
+export def defaultFreedomESDKBuildDir = "build/freedom-e-sdk"
+
+export def defaultFreedomESDKBuildEnv =
+  FreedomESDKBuildEnv defaultFreedomESDKResources defaultFreedomESDKBuildDir
+
+#
+# Virtualenv Creation
+#
+# Freedom E SDK depends on a number of Python libraries. Here we enable
+# the creation of a Virtualenv relative to the BuildDir of Freedom E SDK's
+# Build Environment. By default, these wake rules will ensure the creation
+# of this Virtualenv at run-time with `installFreedomESDKVirtualenv`.
+#
+# If your use-case requires that dependencies are fetched from the internet
+# at workspace creation time, make sure to add `installFreedomESDKVirualenv`
+# to the `preinstall` topic, substituting the FreedomESDKBuildEnv you want
+# to use.
+#
+
+export tuple FreedomESDKVirtualenv =
+  export InstalledFiles: List Path
+
+export def freedomESDKVenvDir buildEnv = "{buildEnv.getFreedomESDKBuildEnvBuildDir}/venv"
+
+def makeFreedomESDKVirtualenvJob buildEnv =
+  def runDir = here
+
+  def inputs =
+    source "freedom-e-sdk/scripts/virtualenv.mk",
+    source "freedom-e-sdk/requirements.txt",
+    mkdir (freedomESDKVenvDir buildEnv),
+    Nil
+
+  def cmd =
+    def venvDir = relative runDir (freedomESDKVenvDir buildEnv)
+    # Explicitly create the venv with the 'python3.7' binary so that we use
+    # the version of python with pip and venv preinstalled in all build environments.
+    "bash", "-c", "%
+      export FREEDOM_E_SDK_VENV_PATH=$(pwd)/%{venvDir}
+      python3.7 -m venv ${FREEDOM_E_SDK_VENV_PATH}
+      . ${FREEDOM_E_SDK_VENV_PATH}/bin/activate && python3 -m pip install pip==20.0.1 --disable-pip-version-check
+      . ${FREEDOM_E_SDK_VENV_PATH}/bin/activate && python3 -m pip install -r requirements.txt
+      touch ${FREEDOM_E_SDK_VENV_PATH}/.stamp
+      %", Nil
+
+  makePlan cmd inputs
+  | setPlanDirectory runDir
+  | setPlanResources ("python/python/3.7.1", Nil)
+  | setPlanKeep True
+
+# Wake rules which depend upon the Freedom E SDK virtualenv should
+# add `installFreedomESDKVirtualenv` to their Plan inputs.
+export target installFreedomESDKVirtualenv buildEnv =
+  def installedFiles =
+    makeFreedomESDKVirtualenvJob buildEnv
+    | runJob
+    | getJobOutputs
+  FreedomESDKVirtualenv installedFiles
+
+#
+# Freedom E SDK Installation
+#
+
+export tuple FreedomESDK =
+  export Dir:            String
+  export InstalledFiles: List Path
+  export BuildEnv: FreedomESDKBuildEnv
+
+# makeFreedomESDK assembles the components of Freedom E SDK into the directory
+# hierarchy expected by the SDK from their flat layout in the wit workspace.
```

Comment:
This particular rule allows you to build another freedom e sdk somewhere else, like in an IP delivery.

Reply:
Right, I was probably unclear in my wording, but I actually meant to comment on the location of the source files, not the destination files. We're making a push make the Wake rules agnostic of the location of the Git repositories, so it will not be safe to assume that all Git repos are at the workspace root.